lib/iris/cube.py

```diff
@@ -2205,12 +2205,11 @@ def new_cell_measure_dims(cm_):
             first_slice = next(slice_gen)
         except StopIteration:
             first_slice = None
-        if self._numpy_array is not None:
-            cube_data = self._numpy_array
-        elif self._dask_array is not None:
+
+        if self.has_lazy_data():
             cube_data = self._dask_array
         else:
-            raise ValueError('This cube has no data, slicing is not supported')
+            cube_data = self._numpy_array
```

Comment:
@dkillick At this point, we're confident that either `self._dask_array` or `self._numpy_array` is populated.

Prior to this change, there seems to be a hint of that might not be the case, hence the `ValueError` exception being raised. I'd certainly hope that there is a data payload of some kind to process - we need to make sure that internally that is always the case.